src/ldb_modules/memberof.c

```diff
@@ -2920,7 +2920,9 @@ static int memberof_mod(struct ldb_module *module, struct ldb_request *req)
     mod_ctx->ghel = ldb_msg_find_element(mod_ctx->msg, DB_GHOST);
 
     /* continue with normal ops if there are no members and no ghosts */
-    if (mod_ctx->membel == NULL && mod_ctx->ghel == NULL) {
+    if ((mod_ctx->membel == NULL && mod_ctx->ghel == NULL) ||
+        (mod_ctx->membel == NULL && mod_ctx->ghel != NULL &&
+         mod_ctx->ghel->num_values == 0)) {
         mod_ctx->terminate = true;
```

Comment:
There is no reason. I just took patches from another branch. I am still seeking optimal workflow with GitHub.

The fixed version is pushed.